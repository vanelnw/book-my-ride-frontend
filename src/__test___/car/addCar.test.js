import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import AddCar from '../../components/car/AddCar';

describe('Test adding new investigator', () => {
  let carAdd;

  beforeAll(async () => {
    carAdd = renderer
      .create(
        <Provider store={store}>
          <Router>
            <AddCar />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('It should render correctly', () => {
    expect(carAdd).toBeTruthy();
  });

  it('Should match its snapshot', () => {
    expect(carAdd).toMatchSnapshot();
  });

  it('It renders the form', () => {
    // to add the data-testid attribute for the form
    const form = screen.findByTestId('add-form');
    waitFor(() => expect(form).toBeInTheDocument());
  });

  it('Should have a button with Add text', () => {
    // to add the data-testid attribute for the button
    const btn = screen.findByTestId('add-btn');
    waitFor(() => expect(btn).toHaveTextContent('Add Car'));
  });
});
