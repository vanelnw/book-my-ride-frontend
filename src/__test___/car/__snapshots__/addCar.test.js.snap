// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test adding new investigator Should match its snapshot 1`] = `
<div
  className="addForm"
>
  <div
    className="innercontainer"
  >
    <div
      className="add-heading"
    >
      <h2>
        Add a new car
      </h2>
    </div>
    <form
      className="add-form"
      onSubmit={[Function]}
    >
      <label
        className="add-form-field"
        htmlFor="make"
      >
        Make:
        <input
          className="addInput"
          id="make"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="add-form-field"
        htmlFor="model"
      >
        Model:
        <input
          className="addInput"
          id="model"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="add-form-field"
        htmlFor="year"
      >
        Year:
        <input
          className="addInput"
          id="year"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="add-form-field"
        htmlFor="image"
      >
        Image:
        <input
          id="image"
          onChange={[Function]}
          type="file"
          value=""
        />
      </label>
      <label
        className="add-form-field"
        htmlFor="description"
      >
        Description:
        <textarea
          id="description"
          onChange={[Function]}
          value=""
        />
      </label>
      <label
        className="add-form-field"
        htmlFor="daily-rate"
      >
        Daily Rate:
        <input
          id="daily-rate"
          onChange={[Function]}
          type="number"
          value=""
        />
      </label>
      <button
        className="btn-add-car"
        type="submit"
      >
        Add Car
      </button>
    </form>
  </div>
</div>
`;
